src/services/accountServices.js

```diff
@@ -0,0 +1,343 @@
+    return { status: false, message: 'Minimum amount error' };
+  }
+
+  const accountEntryResponse = await insertIntoAccounts(accountNumber, 'SAVINGS', id, formattedDate, amount);
+  if (accountEntryResponse.rowCount === 0) {
+    return { status: false, message: 'Unable to open account' };
+  }
+
+  const atmResult = await issueAtmCard(accountNumber);
+  if (atmResult.rowCount === 0) {
+    return { status: false, message: 'Unable to open account' };
+  }
+  return { status: true, message: 'Account Created' };
+};
+
+const createCurrentAccount = async (id, amount, accountNumber, formattedDate) => {
+  if (amount < 100000) {
+    return { status: false, message: 'Minimum amount error' };
+  }
+
+  const accountEntryResponse = insertIntoAccounts(accountNumber, 'CURRENT', id, formattedDate, amount);
+  if (accountEntryResponse.rowCount === 0) {
+    return { status: false, message: 'Unable to open account' };
+  }
+  return { status: true, message: 'Account Created' };
+};
+
+const createLoanAccount = async (id, amount, accountNumber, formattedDate, loanType, duration) => {
+  // Checking for other accounts
+  const userRows = await fetchUserAccounts(id);
+
+  // Checking all the conditions for creating the loan account
+  if (!userRows[0]) {
+    return { status: false, message: 'No other bank account exists' };
+  }
+  if (amount < 500000) {
+    return { status: false, message: 'Minimum amount error' };
+  }
+
+  // Can only give 40% of total deposit as loan
+  const totalSum = getTotalDepositsOfUser(userRows);
+  if (((totalSum * 40) / 100) < amount) {
+    return { status: false, message: 'Loan amount should be lesser than 40% of total deposits' };
+  }
+
+  const loanInterestResponse = getLoanInterest(loanType);
+  if (!loanInterestResponse.status) {
+    return { status: false, message: 'Invalid details entered' };
+  }
+  const { loanInterest } = loanInterestResponse;
+
+  const accountEntryResponse = await insertIntoAccounts(accountNumber, 'LOAN', id, formattedDate, amount);
+  const loanAccountEntryResponse = await insertIntoLoanAccounts(accountNumber, loanType, loanInterest, amount, duration, 'active');
+
+  if (accountEntryResponse.rowCount === 0 || (await loanAccountEntryResponse).rowCount === 0) {
+    return { status: false, message: 'Loan cannot be provided' };
+  }
+  return { status: true, message: 'Account Created' };
+};
+
+const createBankAccount = async (req) => {
+  const { id, accountType, amount } = req;
+  const userDetails = await getUserDetails(id);
+  const { dob } = userDetails;
+
+  // Checking if the same accountType already exists for the current user
+  const accountResponse = await getUserAccount(id, accountType);
+  if (accountResponse) {
+    return { status: false, message: 'Account already exists' };
+  }
+
+  const accountNumber = BigInt(generateAccountNo(15));
+  const currentAge = calculateAge(dob);
+
+  if (accountType !== 'SAVINGS' && accountType !== 'CURRENT' && accountType !== 'LOAN') {
```

Comment:
can we do something like this ? `var validAccountType = ["SAVINGS", "CURRENT", "LOAN"]` and see if the req.account_type is in the array